matrix-sdk/src/main/java/org/matrix/androidsdk/MXSession.java

```diff
@@ -1198,74 +1196,60 @@ public boolean createDirectMessageRoom(final String aParticipantUserId, final St
             params.setDirectMessage();
             params.addParticipantIds(mHsConfig, Arrays.asList(aParticipantUserId));
 
-            createRoom(params, new ApiCallback<String>() {
-                @Override
-                public void onSuccess(String roomId) {
-                    final String fRoomId = roomId;
-                    toggleDirectChatRoom(roomId, aParticipantUserId, new ApiCallback<Void>() {
-                        @Override
-                        public void onSuccess(Void info) {
-                            Room room = getDataHandler().getRoom(fRoomId);
-
-                            if (null != room) {
-                                room.markAllAsRead(null);
-                            }
+            createRoom(params, aCreateRoomCallBack);
+        }
 
-                            if (null != aCreateRoomCallBack) {
-                                aCreateRoomCallBack.onSuccess(fRoomId);
-                            }
-                        }
+        return retCode;
+    }
 
-                        @Override
-                        public void onNetworkError(Exception e) {
-                            if (null != aCreateRoomCallBack) {
-                                aCreateRoomCallBack.onNetworkError(e);
-                            }
-                        }
+    /**
+     * Set the created room as a direct message one.
+     *
+     * @param roomId   the room id
+     * @param userIds  the user ids list
+     * @param callback the asynchronous callback
+     */
+    private void setDirectMessageRoom(final String roomId, final List<String> userIds, final ApiCallback<String> callback) {
```

Comment:
I would rename this method with "finalizeDirectMessageRoom" because it is limited to the room creation.